diff --git a/spec/Entity/DinosaurSpec.php b/spec/Entity/DinosaurSpec.php
index e1ea532..06884ea 100644
--- a/spec/Entity/DinosaurSpec.php
+++ b/spec/Entity/DinosaurSpec.php
@@ -76,4 +76,16 @@ class DinosaurSpec extends ObjectBehavior
         $this->getGenus()->shouldBe('Velociraptor');
         $this->getLength()->shouldBe(5);
     }
+
+    public function it_should_be_herbavore_by_default()
+    {
+        $this->shouldNotBeCarnivorous();
+    }
+
+    public function it_should_allow_to_check_if_dinosaur_is_carnivorous()
+    {
+        $this->beConstructedWith('Velociraptor', true);
+
+        $this->shouldBeCarnivorous();
+    }
 }
diff --git a/src/Entity/Dinosaur.php b/src/Entity/Dinosaur.php
index a863148..fbdb77b 100644
--- a/src/Entity/Dinosaur.php
+++ b/src/Entity/Dinosaur.php
@@ -55,4 +55,14 @@ class Dinosaur
     {
         return $this->genus;
     }
+
+    public function isCarnivorous(): bool
+    {
+        return $this->isCarnivorous;
+    }
+
+    public function hasSameDietAs(Dinosaur $dinosaur): bool
+    {
+        return $this->isCarnivorous() === $dinosaur->isCarnivorous();
+    }
 }
diff --git a/src/Entity/Enclosure.php b/src/Entity/Enclosure.php
index 11e6cd0..2264baa 100644
--- a/src/Entity/Enclosure.php
+++ b/src/Entity/Enclosure.php
@@ -2,6 +2,8 @@
 
 namespace App\Entity;
 
+use App\Exception\NotABuffetException;
+
 class Enclosure
 {
     /** @var Dinosaur[]|iterable */
@@ -14,6 +16,15 @@ class Enclosure
 
     public function addDinosaur(Dinosaur $dinosaur)
     {
+        if (!$this->canAddDinosaur($dinosaur)) {
+            throw new NotABuffetException();
+        }
+
         $this->dinosaurs[] = $dinosaur;
     }
+
+    private function canAddDinosaur(Dinosaur $dinosaur): bool
+    {
+        return \count($this->dinosaurs) === 0 || $this->dinosaurs[0]->hasSameDietAs($dinosaur);
+    }
 }
