diff --git a/spec/Entity/EnclosureSpec.php b/spec/Entity/EnclosureSpec.php
index cf5795b..192fae5 100644
--- a/spec/Entity/EnclosureSpec.php
+++ b/spec/Entity/EnclosureSpec.php
@@ -23,6 +23,8 @@ class EnclosureSpec extends ObjectBehavior
 
     function it_should_be_able_to_add_dinosaurs()
     {
+        $this->beConstructedWith(true);
+
         $this->addDinosaur(new Dinosaur());
         $this->addDinosaur(new Dinosaur());
 
@@ -31,6 +33,8 @@ class EnclosureSpec extends ObjectBehavior
 
     function it_should_not_allow_to_add_carnivorous_dinosaurs_to_non_carnivorous_enclosure()
     {
+        $this->beConstructedWith(true);
+
         $this->addDinosaur(new Dinosaur('veggie-eater', false));
 
         $this
@@ -40,6 +44,8 @@ class EnclosureSpec extends ObjectBehavior
 
     function it_should_not_allow_to_add_dinosaurs_to_unsecure_enclosures()
     {
+        $this->beConstructedWith(false);
+
         $this
             ->shouldThrow(new DinosaursAreRunningRampantException('Are you craaazy?!?'))
             ->duringAddDinosaur(new Dinosaur('Velociraptor', true));
diff --git a/src/Entity/Enclosure.php b/src/Entity/Enclosure.php
index 283bd80..dc29be6 100644
--- a/src/Entity/Enclosure.php
+++ b/src/Entity/Enclosure.php
@@ -13,6 +13,13 @@ class Enclosure
     /** @var Security[] */
     private $securities = [];
 
+    public function __construct(bool $withBasicSecurity = false)
+    {
+        if ($withBasicSecurity) {
+            $this->addSecurity(new Security('Fence', true, $this));
+        }
+    }
+
     public function getDinosaurs(): array
     {
         return $this->dinosaurs;
@@ -31,6 +38,11 @@ class Enclosure
         $this->dinosaurs[] = $dinosaur;
     }
 
+    public function addSecurity(Security $security)
+    {
+        $this->securities[] = $security;
+    }
+
     private function canAddDinosaur(Dinosaur $dinosaur): bool
     {
         return count($this->dinosaurs) === 0 || $dinosaur->hasSameDietAs($this->dinosaurs[0]);
