diff --git a/spec/Entity/DinosaurSpec.php b/spec/Entity/DinosaurSpec.php
index 2686495..d1b7b33 100644
--- a/spec/Entity/DinosaurSpec.php
+++ b/spec/Entity/DinosaurSpec.php
@@ -76,4 +76,16 @@ class DinosaurSpec extends ObjectBehavior
         $this->getGenus()->shouldBe('Velociraptor');
         $this->getLength()->shouldBe(5);
     }
+
+    function it_should_be_herbivore_by_default()
+    {
+        $this->shouldNotBeCarnivorous();
+    }
+
+    function it_should_allow_to_check_if_dinosaur_is_carnivorous()
+    {
+        $this->beConstructedWith('Velociraptor', true);
+
+        $this->shouldBeCarnivorous();
+    }
 }
diff --git a/src/Entity/Dinosaur.php b/src/Entity/Dinosaur.php
index 328ed00..751d2cc 100644
--- a/src/Entity/Dinosaur.php
+++ b/src/Entity/Dinosaur.php
@@ -48,4 +48,9 @@ class Dinosaur
     {
         return $this->genus;
     }
+
+    public function isCarnivorous()
+    {
+        // TODO: write logic here
+    }
 }
diff --git a/src/Entity/Enclosure.php b/src/Entity/Enclosure.php
index 38c7d7b..f5c063a 100644
--- a/src/Entity/Enclosure.php
+++ b/src/Entity/Enclosure.php
@@ -2,6 +2,8 @@
 
 namespace App\Entity;
 
+use App\Exception\NotABuffetException;
+
 class Enclosure
 {
     /** @var Dinosaur[] */
@@ -14,6 +16,16 @@ class Enclosure
 
     public function addDinosaur(Dinosaur $dinosaur)
     {
+        if (!$this->canAddDinosaur($dinosaur)) {
+            throw new NotABuffetException();
+        }
+
         $this->dinosaurs[] = $dinosaur;
     }
+
+    private function canAddDinosaur(Dinosaur $dinosaur): bool
+    {
+        return count($this->dinosaurs) === 0 ||
+            ($this->dinosaurs[0]->isCarnivorous() === $dinosaur->isCarnivorous());
+    }
 }
