diff --git a/src/Entity/Enclosure.php b/src/Entity/Enclosure.php
index dc29be6..21966c6 100644
--- a/src/Entity/Enclosure.php
+++ b/src/Entity/Enclosure.php
@@ -13,11 +13,15 @@ class Enclosure
     /** @var Security[] */
     private $securities = [];
 
-    public function __construct(bool $withBasicSecurity = false)
+    public function __construct(bool $withBasicSecurity = false, array $initialDinosaurs = [])
     {
         if ($withBasicSecurity) {
             $this->addSecurity(new Security('Fence', true, $this));
         }
+
+        foreach ($initialDinosaurs as $dinosaur) {
+            $this->addDinosaur($dinosaur);
+        }
     }
 
     public function getDinosaurs(): array
