diff --git a/src/Entity/Enclosure.php b/src/Entity/Enclosure.php
index 2264baa..4dc8069 100644
--- a/src/Entity/Enclosure.php
+++ b/src/Entity/Enclosure.php
@@ -2,6 +2,7 @@
 
 namespace App\Entity;
 
+use App\Exception\DinosaursAreRunningRampantException;
 use App\Exception\NotABuffetException;
 
 class Enclosure
@@ -9,6 +10,9 @@ class Enclosure
     /** @var Dinosaur[]|iterable */
     private $dinosaurs = [];
 
+    /** @var Security[]|iterable */
+    private $securities = [];
+
     public function getDinosaurs(): iterable
     {
         return $this->dinosaurs;
@@ -16,6 +20,10 @@ class Enclosure
 
     public function addDinosaur(Dinosaur $dinosaur)
     {
+        if (!$this->getIsSecurityActive()) {
+            throw new DinosaursAreRunningRampantException('Are you craaazy?!?');
+        }
+
         if (!$this->canAddDinosaur($dinosaur)) {
             throw new NotABuffetException();
         }
@@ -27,4 +35,15 @@ class Enclosure
     {
         return \count($this->dinosaurs) === 0 || $this->dinosaurs[0]->hasSameDietAs($dinosaur);
     }
+
+    private function getIsSecurityActive(): bool
+    {
+        foreach ($this->securities as $security) {
+            if ($security->getIsActive()) {
+                return true;
+            }
+        }
+
+        return false;
+    }
 }
diff --git a/src/Entity/Security.php b/src/Entity/Security.php
new file mode 100644
index 0000000..7ee1a41
--- /dev/null
+++ b/src/Entity/Security.php
@@ -0,0 +1,34 @@
+<?php
+
+namespace App\Entity;
+
+class Security
+{
+    /**
+     * @var string
+     */
+    private $name;
+
+    /**
+     * @var bool
+     */
+    private $isActive;
+
+    /**
+     * @var Enclosure
+     */
+    private $enclosure;
+
+    public function __construct(string $name, bool $isActive, Enclosure $enclosure)
+    {
+
+        $this->name = $name;
+        $this->isActive = $isActive;
+        $this->enclosure = $enclosure;
+    }
+
+    public function getIsActive(): bool
+    {
+        return $this->isActive;
+    }
+}
