diff --git a/spec/Matcher/BeGreaterMatcher.php b/spec/Matcher/BeGreaterMatcher.php
index cf237ea..9c9a479 100644
--- a/spec/Matcher/BeGreaterMatcher.php
+++ b/spec/Matcher/BeGreaterMatcher.php
@@ -2,7 +2,28 @@
 
 namespace spec\Matcher;
 
-class BeGreaterMatcher
+use PhpSpec\Exception\Example\FailureException;
+use PhpSpec\Matcher\BasicMatcher;
+
+final class BeGreaterMatcher extends BasicMatcher
 {
+    protected function matches($subject, array $arguments): bool
+    {
+        // TODO: Implement matches() method.
+    }
+
+    protected function getFailureException(string $name, $subject, array $arguments): FailureException
+    {
+        // TODO: Implement getFailureException() method.
+    }
+
+    protected function getNegativeFailureException(string $name, $subject, array $arguments): FailureException
+    {
+        // TODO: Implement getNegativeFailureException() method.
+    }
 
+    public function supports(string $name, $subject, array $arguments): bool
+    {
+        // TODO: Implement supports() method.
+    }
 }
