diff --git a/src/Factory/DinosaurFactory.php b/src/Factory/DinosaurFactory.php
index c125d1c..3fb0f35 100644
--- a/src/Factory/DinosaurFactory.php
+++ b/src/Factory/DinosaurFactory.php
@@ -2,10 +2,15 @@
 
 namespace App\Factory;
 
+use App\Entity\Dinosaur;
+
 class DinosaurFactory
 {
-    public function growVelociraptor($argument1)
+    public function growVelociraptor(int $length): Dinosaur
     {
-        // TODO: write logic here
+        $dinosaur = new Dinosaur('Velociraptor', true);
+        $dinosaur->setLength($length);
+
+        return $dinosaur;
     }
 }
