diff --git a/src/Entity/Dinosaur.php b/src/Entity/Dinosaur.php
index 3978cba..b77ed05 100644
--- a/src/Entity/Dinosaur.php
+++ b/src/Entity/Dinosaur.php
@@ -6,9 +6,14 @@ class Dinosaur
 {
     private $length = 0;
 
-    public function __construct($argument1, $argument2)
+    private $genus;
+
+    private $isCarnivorous;
+
+    public function __construct(string $genus = 'Unknown', bool $isCarnivorous = false)
     {
-        // TODO: write logic here
+        $this->genus = $genus;
+        $this->isCarnivorous = $isCarnivorous;
     }
 
     public function getLength(): int
@@ -23,6 +28,11 @@ class Dinosaur
 
     public function getDescription(): string
     {
-        return 'The Unknown non-carnivorous dinosaur is 0 meters long';
+        return sprintf(
+            'The %s %scarnivorous dinsaur is %d meters long',
+            $this->genus,
+            $this->isCarnivorous ? '' : 'non-',
+            $this->length
+        );
     }
 }
