diff --git a/spec/Factory/DinosaurFactorySpec.php b/spec/Factory/DinosaurFactorySpec.php
new file mode 100644
index 0000000..2314c58
--- /dev/null
+++ b/spec/Factory/DinosaurFactorySpec.php
@@ -0,0 +1,26 @@
+<?php
+
+namespace spec\App\Factory;
+
+use App\Entity\Dinosaur;
+use App\Factory\DinosaurFactory;
+use PhpSpec\ObjectBehavior;
+use Prophecy\Argument;
+
+class DinosaurFactorySpec extends ObjectBehavior
+{
+    function it_is_initializable()
+    {
+        $this->shouldHaveType(DinosaurFactory::class);
+    }
+
+    function it_grows_a_large_velociraptor()
+    {
+        $dinosaur = $this->growVelociraptor(5);
+
+        $dinosaur->shouldBeAnInstanceOf(Dinosaur::class);
+        $dinosaur->getGenus()->shouldBeString();
+        $dinosaur->getGenus()->shouldBe('Velociraptor');
+        $dinosaur->getLength()->shouldBe(5);
+    }
+}
diff --git a/src/Factory/DinosaurFactory.php b/src/Factory/DinosaurFactory.php
new file mode 100644
index 0000000..c125d1c
--- /dev/null
+++ b/src/Factory/DinosaurFactory.php
@@ -0,0 +1,11 @@
+<?php
+
+namespace App\Factory;
+
+class DinosaurFactory
+{
+    public function growVelociraptor($argument1)
+    {
+        // TODO: write logic here
+    }
+}
