diff --git a/src/Entity/Dinosaur.php b/src/Entity/Dinosaur.php
index b7abcf7..fedf7f0 100644
--- a/src/Entity/Dinosaur.php
+++ b/src/Entity/Dinosaur.php
@@ -16,7 +16,7 @@ class Dinosaur
         $this->length = $length;
     }
 
-    public function getDescription()
+    public function getDescription(): string
     {
         return 'The Unknown non-carnivorous dinosaur is 0 meters long';
     }
