diff --git a/phpspec.yml b/phpspec.yml
index f4131ea..e5a7cd2 100644
--- a/phpspec.yml
+++ b/phpspec.yml
@@ -2,3 +2,6 @@ suites:
     default:
         namespace: App
         psr4_prefix: App
+
+matchers:
+    - spec\Matcher\BeGreaterMatcher
diff --git a/spec/Entity/DinosaurSpec.php b/spec/Entity/DinosaurSpec.php
index b30d11f..74877b2 100644
--- a/spec/Entity/DinosaurSpec.php
+++ b/spec/Entity/DinosaurSpec.php
@@ -46,4 +46,11 @@ class DinosaurSpec extends ObjectBehavior
 
         $this->getLength()->shouldReturn(9);
     }
+
+    function it_should_not_shrink()
+    {
+        $this->setLength(15);
+
+        $this->getLength()->shouldBeGreaterThan(12);
+    }
 }
diff --git a/spec/Matcher/BeGreaterMatcher.php b/spec/Matcher/BeGreaterMatcher.php
new file mode 100644
index 0000000..cf237ea
--- /dev/null
+++ b/spec/Matcher/BeGreaterMatcher.php
@@ -0,0 +1,8 @@
+<?php
+
+namespace spec\Matcher;
+
+class BeGreaterMatcher
+{
+
+}
