diff --git a/src/Factory/DinosaurFactory.php b/src/Factory/DinosaurFactory.php
index a14d94e..d3f0acb 100644
--- a/src/Factory/DinosaurFactory.php
+++ b/src/Factory/DinosaurFactory.php
@@ -11,9 +11,11 @@ class DinosaurFactory
         return $this->createDinosaur('Velociraptor', true, $length);
     }
 
-    private function createDinosaur(string $genus, bool $isCarnivorous, int $length)
+    private function createDinosaur(string $genus, bool $isCarnivorous, int $length): Dinosaur
     {
         $dinosaur = new Dinosaur($genus, $isCarnivorous);
         $dinosaur->setLength($length);
+
+        return $dinosaur;
     }
 }
