diff --git a/src/Factory/DinosaurFactory.php b/src/Factory/DinosaurFactory.php
index 3fb0f35..a14d94e 100644
--- a/src/Factory/DinosaurFactory.php
+++ b/src/Factory/DinosaurFactory.php
@@ -8,9 +8,12 @@ class DinosaurFactory
 {
     public function growVelociraptor(int $length): Dinosaur
     {
-        $dinosaur = new Dinosaur('Velociraptor', true);
-        $dinosaur->setLength($length);
+        return $this->createDinosaur('Velociraptor', true, $length);
+    }
 
-        return $dinosaur;
+    private function createDinosaur(string $genus, bool $isCarnivorous, int $length)
+    {
+        $dinosaur = new Dinosaur($genus, $isCarnivorous);
+        $dinosaur->setLength($length);
     }
 }
