diff --git a/src/Entity/Dinosaur.php b/src/Entity/Dinosaur.php
index e810e16..328ed00 100644
--- a/src/Entity/Dinosaur.php
+++ b/src/Entity/Dinosaur.php
@@ -16,9 +16,12 @@ class Dinosaur
         $this->isCarnivorous = $isCarnivorous;
     }
 
-    public static function growVelociraptor($argument1)
+    public static function growVelociraptor(int $length): self
     {
-        throw new \BadMethodCallException("Mismatch between the number of arguments of the factory method and constructor");
+        $dinosaur = new static('Velociraptor', true);
+        $dinosaur->setLength($length);
+
+        return $dinosaur;
     }
 
     public function getLength(): int
@@ -40,4 +43,9 @@ class Dinosaur
             $this->length
         );
     }
+
+    public function getGenus(): string
+    {
+        return $this->genus;
+    }
 }
