diff --git a/spec/Service/EnclosureBuilderServiceSpec.php b/spec/Service/EnclosureBuilderServiceSpec.php
index 40c9cf0..b15a105 100644
--- a/spec/Service/EnclosureBuilderServiceSpec.php
+++ b/spec/Service/EnclosureBuilderServiceSpec.php
@@ -27,12 +27,15 @@ class EnclosureBuilderServiceSpec extends ObjectBehavior
         $dinosaurFactory->growVelociraptor(Argument::any())->willReturn(
             $dino1,
             $dino2
-        )->shouldBeCalledTimes(2);
+        );
 
         $enclosure = $this->buildEnclosure(1, 2);
 
         $enclosure->shouldBeAnInstanceOf(Enclosure::class);
         $enclosure->getIsSecurityActive()->shouldReturn(true);
         $enclosure->getDinosaurs()[0]->getLength()->shouldReturn(4);
+
+        $dinosaurFactory->growVelociraptor(Argument::any())
+            ->shouldHaveBeenCalledTimes(2);
     }
 }
