diff --git a/spec/Entity/DinosaurSpec.php b/spec/Entity/DinosaurSpec.php
index 99a7e22..b5dea21 100644
--- a/spec/Entity/DinosaurSpec.php
+++ b/spec/Entity/DinosaurSpec.php
@@ -93,4 +93,9 @@ class DinosaurSpec extends ObjectBehavior
     {
         $this->shouldHaveSameDietAs(new Dinosaur());
     }
+
+    function it_should_allow_to_check_if_two_dinosaurs_have_same_diet_using_stub(Dinosaur $dinosaur)
+    {
+        var_dump($dinosaur);
+    }
 }
