diff --git a/spec/Service/EnclosureBuilderServiceSpec.php b/spec/Service/EnclosureBuilderServiceSpec.php
index b8f1d5c..5e6503e 100644
--- a/spec/Service/EnclosureBuilderServiceSpec.php
+++ b/spec/Service/EnclosureBuilderServiceSpec.php
@@ -26,7 +26,7 @@ class EnclosureBuilderServiceSpec extends ObjectBehavior
         $dinosaurFactory->growVelociraptor(Argument::type('integer'))->willReturn(
             $dino1,
             $dino2
-        )->shouldBeCalledTimes(2);
+        );
 
         $enclosure = $this->buildEnclosure(1, 2);
 
@@ -34,5 +34,8 @@ class EnclosureBuilderServiceSpec extends ObjectBehavior
         $enclosure->isSecurityActive()->shouldReturn(true);
         $enclosure->getDinosaurs()[0]->shouldBe($dino1);
         $enclosure->getDinosaurs()[1]->shouldBe($dino2);
+
+        $dinosaurFactory->growVelociraptor(Argument::any())
+            ->shouldHaveBeenCalledTimes(2);
     }
 }
