diff --git a/spec/Service/EnclosureBuilderServiceSpec.php b/spec/Service/EnclosureBuilderServiceSpec.php
new file mode 100644
index 0000000..4daec5a
--- /dev/null
+++ b/spec/Service/EnclosureBuilderServiceSpec.php
@@ -0,0 +1,39 @@
+<?php
+
+namespace spec\App\Service;
+
+use App\Entity\Dinosaur;
+use App\Entity\Enclosure;
+use App\Factory\DinosaurFactory;
+use App\Service\EnclosureBuilderService;
+use App\Service\EntityManagerInterface;
+use PhpSpec\ObjectBehavior;
+use Prophecy\Argument;
+
+class EnclosureBuilderServiceSpec extends ObjectBehavior
+{
+    function let(EntityManagerInterface $entityManager, DinosaurFactory $dinosaurFactory)
+    {
+        $this->beConstructedWith($entityManager, $dinosaurFactory);
+
+        $dinosaurFactory->growVelociraptor(Argument::any())->willReturn(
+            new Dinosaur('Velociraptor', 4),
+            new Dinosaur('Velociraptor', 8)
+        );
+    }
+
+    function it_is_initializable()
+    {
+        $this->shouldHaveType(EnclosureBuilderService::class);
+    }
+
+    function it_persists_built_enclosure(EntityManagerInterface $entityManager, DinosaurFactory $dinosaurFactory)
+    {
+        $entityManager->persist(Argument::type(Enclosure::class))->shouldBeCalled();
+        $entityManager->flush()->shouldBeCalled();
+
+        $this->buildEnclosure(1, 2);
+
+        $dinosaurFactory->growVelociraptor(Argument::any())->shouldHaveBeenCalledTimes(2);
+    }
+}
diff --git a/src/Service/EnclosureBuilderService.php b/src/Service/EnclosureBuilderService.php
new file mode 100644
index 0000000..d6939d5
--- /dev/null
+++ b/src/Service/EnclosureBuilderService.php
@@ -0,0 +1,16 @@
+<?php
+
+namespace App\Service;
+
+class EnclosureBuilderService
+{
+    public function __construct($argument1, $argument2)
+    {
+        // TODO: write logic here
+    }
+
+    public function buildEnclosure($argument1, $argument2)
+    {
+        // TODO: write logic here
+    }
+}
diff --git a/src/Service/EntityManagerInterface.php b/src/Service/EntityManagerInterface.php
new file mode 100644
index 0000000..89dd277
--- /dev/null
+++ b/src/Service/EntityManagerInterface.php
@@ -0,0 +1,10 @@
+<?php
+
+namespace App\Service;
+
+interface EntityManagerInterface
+{
+    public function persist($object);
+
+    public function flush();
+}
