diff --git a/src/Service/EnclosureBuilderService.php b/src/Service/EnclosureBuilderService.php
index daf4979..10b400e 100644
--- a/src/Service/EnclosureBuilderService.php
+++ b/src/Service/EnclosureBuilderService.php
@@ -2,10 +2,31 @@
 
 namespace App\Service;
 
+use App\Entity\Enclosure;
+use App\Entity\Security;
+
 class EnclosureBuilderService
 {
-    public function buildEnclosure($argument1, $argument2)
+    public function buildEnclosure(
+        int $numberOfSecuritySystems = 1,
+        int $numberOfDinosaurs = 3
+    ): Enclosure
     {
-        // TODO: write logic here
+        $enclosure = new Enclosure();
+
+        $this->addSecuritySystems($numberOfSecuritySystems, $enclosure);
+
+        return $enclosure;
+    }
+
+    private function addSecuritySystems(int $numberOfSecuritySystems, Enclosure $enclosure)
+    {
+        $securityNames = ['Fence', 'Electric fence', 'Guard tower'];
+        for ($i = 0; $i < $numberOfSecuritySystems; $i++) {
+            $securityName = $securityNames[array_rand($securityNames)];
+            $security = new Security($securityName, true, $enclosure);
+
+            $enclosure->addSecurity($security);
+        }
     }
 }
