diff --git a/spec/Service/EnclosureBuilderServiceSpec.php b/spec/Service/EnclosureBuilderServiceSpec.php
index 5e6503e..4afb940 100644
--- a/spec/Service/EnclosureBuilderServiceSpec.php
+++ b/spec/Service/EnclosureBuilderServiceSpec.php
@@ -11,6 +11,11 @@ use Prophecy\Argument;
 
 class EnclosureBuilderServiceSpec extends ObjectBehavior
 {
+    function let(DinosaurFactory $dinosaurFactory)
+    {
+        $this->beConstructedWith($dinosaurFactory);
+    }
+
     function it_is_initializable()
     {
         $this->shouldHaveType(EnclosureBuilderService::class);
@@ -18,7 +23,6 @@ class EnclosureBuilderServiceSpec extends ObjectBehavior
 
     function it_builds_enclosure_with_dinosaurs(DinosaurFactory $dinosaurFactory)
     {
-        $this->beConstructedWith($dinosaurFactory);
         $dino1 = new Dinosaur('Stegosaurus', false);
         $dino1->setLength(6);
         $dino2 = new Dinosaur('Baby Stegosaurus', false);
