diff --git a/spec/Service/EnclosureBuilderServiceSpec.php b/spec/Service/EnclosureBuilderServiceSpec.php
index 4afb940..53412e8 100644
--- a/spec/Service/EnclosureBuilderServiceSpec.php
+++ b/spec/Service/EnclosureBuilderServiceSpec.php
@@ -6,14 +6,15 @@ use App\Entity\Dinosaur;
 use App\Entity\Enclosure;
 use App\Factory\DinosaurFactory;
 use App\Service\EnclosureBuilderService;
+use App\Service\EntityManagerInterface;
 use PhpSpec\ObjectBehavior;
 use Prophecy\Argument;
 
 class EnclosureBuilderServiceSpec extends ObjectBehavior
 {
-    function let(DinosaurFactory $dinosaurFactory)
+    function let(DinosaurFactory $dinosaurFactory, EntityManagerInterface $entityManager)
     {
-        $this->beConstructedWith($dinosaurFactory);
+        $this->beConstructedWith($dinosaurFactory, $entityManager);
     }
 
     function it_is_initializable()
@@ -21,7 +22,7 @@ class EnclosureBuilderServiceSpec extends ObjectBehavior
         $this->shouldHaveType(EnclosureBuilderService::class);
     }
 
-    function it_builds_enclosure_with_dinosaurs(DinosaurFactory $dinosaurFactory)
+    function it_builds_enclosure_with_dinosaurs(DinosaurFactory $dinosaurFactory, EntityManagerInterface $entityManager)
     {
         $dino1 = new Dinosaur('Stegosaurus', false);
         $dino1->setLength(6);
@@ -41,5 +42,9 @@ class EnclosureBuilderServiceSpec extends ObjectBehavior
 
         $dinosaurFactory->growVelociraptor(Argument::any())
             ->shouldHaveBeenCalledTimes(2);
+
+        $entityManager->persist(Argument::type(Enclosure::class))
+            ->shouldHaveBeenCalled();
+        $entityManager->flush()->shouldHaveBeenCalled();
     }
 }
diff --git a/src/Service/EntityManagerInterface.php b/src/Service/EntityManagerInterface.php
new file mode 100644
index 0000000..89dd277
--- /dev/null
+++ b/src/Service/EntityManagerInterface.php
@@ -0,0 +1,10 @@
+<?php
+
+namespace App\Service;
+
+interface EntityManagerInterface
+{
+    public function persist($object);
+
+    public function flush();
+}
