diff --git a/spec/Service/EnclosureBuilderServiceSpec.php b/spec/Service/EnclosureBuilderServiceSpec.php
index 1236182..fe4b18c 100644
--- a/spec/Service/EnclosureBuilderServiceSpec.php
+++ b/spec/Service/EnclosureBuilderServiceSpec.php
@@ -23,18 +23,16 @@ class EnclosureBuilderServiceSpec extends ObjectBehavior
         $dino1->setLength(6);
         $dino2 = new Dinosaur('Baby Stegosaurus', false);
         $dino2->setLength(2);
-        $dinosaurFactory->growVelociraptor(5)->willReturn(
-            $dino2
-        );
         $dinosaurFactory->growVelociraptor(Argument::type('integer'))->willReturn(
-            $dino1
+            $dino1,
+            $dino2
         );
 
         $enclosure = $this->buildEnclosure(1, 2);
 
         $enclosure->shouldBeAnInstanceOf(Enclosure::class);
         $enclosure->isSecurityActive()->shouldReturn(true);
-        $enclosure->getDinosaurs()[0]->shouldBe($dino2);
-        $enclosure->getDinosaurs()[1]->shouldBe($dino1);
+        $enclosure->getDinosaurs()[0]->shouldBe($dino1);
+        $enclosure->getDinosaurs()[1]->shouldBe($dino2);
     }
 }
